### PR TITLE
nixos/hyprland: enable xdg.portal by default + use lib.mkDefault for enableWlrPortal

### LINKED ISSUE #319630: Sway and Hyprland modules incompatible
### Describe the bug
It is impossible to have both sway and hyprland enabled at the same time.

### Steps To Reproduce

```
programs.sway.enable = true;
programs.hyprland.enable = true;
```

then nixos-rebuild results in:

```
...
error: The option `xdg.portal.wlr.enable' has conflicting definition values:
       - In `/nix/store/dydg48djlykksz8cxq0xjplyxpa9pvf4-source/nixos/modules/programs/wayland/sway.nix': true
       - In `/nix/store/dydg48djlykksz8cxq0xjplyxpa9pvf4-source/nixos/modules/programs/wayland/hyprland.nix': false
       Use `lib.mkForce value` or `lib.mkDefault value` to change the priority on any of these definitions.
```

### Expected behavior
Have a system with both sway and hyprland available simultaneously.

### Additional context

Caused by #315827.

### Notify maintainers

@wozeparrot @fufexan 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.1, NixOS, 24.05 (Uakari), 24.05.20240517.4a6b83b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/0aavdx9m5ms1cj5pb1dx0brbrbigy8ij-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by leon-erd**
## Description of changes

The hyprland module specifies 
```nix
xdg.portal = {
  extraPortals = [ cfg.portalPackage ];
  configPackages = lib.mkDefault [ cfg.package ];
};
```
but doesn't set `xdg.portal.enable = true;`. I think it was previously set to `true` because of `xdg.portal.wlr.enable = true;`.
However, since this MR https://github.com/NixOS/nixpkgs/pull/315827 sets `xdg.portal.wlr.enable = false;` by default xdg-dektop-portal will be disabled when using the hyprland module out of the box.

Fixes https://github.com/NixOS/nixpkgs/issues/319630

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---
@JohnRTitor @fufexan

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/programs/wayland/hyprland.nix [hunk 1]
```diff
@@ -56,6 +56,7 @@ in
       services.displayManager.sessionPackages = [ cfg.package ];
 
       xdg.portal = {
+        enable = lib.mkDefault true;
```
Comment (SuperSandro2000) [inline nixos/modules/programs/wayland/hyprland.nix:59]:
```suggestion
        enable = true;
```
we probably want to confirm users when they are disabling that and that it isn't the recommended setup

Comment (JohnRTitor) [reply nixos/modules/programs/wayland/hyprland.nix:59]: I agree, and eval for some reason is failing?

**Review by JohnRTitor — CHANGES_REQUESTED** [review]
Please squash the commits and force-push.

You might want to rebase the branch to latest master as well, since eval was failing earlier.

**Comment by leon-erd** [comment]
Done! Eval also seems to work now :)

**Review by JohnRTitor — APPROVED** [review]
LGTM

**Review by fufexan — APPROVED** [review]

**Comment by leon-erd** [comment]
Is there anything more that I need to do to get this merged? :)
@SuperSandro2000 do you have merging rights or should I ping someone else?

**Review by donovanglover — COMMENTED** [review]
Could make the commit easier to grep for by using `nixos/hyprland:` instead of `hyprland module:`. See the [file history](https://github.com/NixOS/nixpkgs/commits/release-24.05/nixos/modules/programs/wayland/hyprland.nix).

**Comment by leon-erd** [comment]
Sure! Do I have to ping anyone specific?

**Comment by fufexan** [comment]
@leon-erd can you also update the commit name? Easiest method is `git commit --amend` and editing the message.

**Comment by fufexan** [comment]
Can you also set `enableWlrPortal` to `lib.mkDefault false;`? That would fix https://github.com/NixOS/nixpkgs/issues/319630.

**Comment by leon-erd** [comment]
Do you want me to squash the commits?

**Comment by fufexan** [comment]
Nope, they're fine separate. Also please add "fixes https://github.com/NixOS/nixpkgs/issues/319630" in the PR description.

**Review by fufexan — APPROVED** [review]

**Review by donovanglover — APPROVED** [review]

**Comment by Atemu** [comment]
Ran into this issue, worked around it in my config and then ran into this PR while looking for something else.

Thanks!

**Comment by fufexan** [comment]
@JohnRTitor I haven't seen any changes in functionality, but I also don't use xdg autostart.

**Pull request merged**